pkg/estimator/client/general.go

```diff
@@ -32,15 +32,17 @@ import (
 
 // GeneralEstimator is the default replica estimator.
 func init() {
-	replicaEstimators["general-estimator"] = NewGeneralEstimator()
+	replicaEstimators = append(replicaEstimators, NewGeneralEstimator(General))
```

Comment:
It's upset that golang does not have data structure like rbtree, otherwise we don't need to change `replicaEstimators` to slice.

Reply:
feel upset too

Reply:
good idea, kindly refer [PR](https://github.com/karmada-io/karmada/pull/4555) about rbTree.

Reply:
@whitewindmills If we change `replicaEstimators []ReplicaEstimator` to `replicaEstimators *rbt.Tree`,

how can I prevent `func RegisterSchedulerEstimator(se *SchedulerEstimator)` function been called multiple times in `descheduler_test.go`

Reply:
Beside, it can not store object in same priority if the key is priority:

```go
tree := rbt.NewWithIntComparator() // empty (keys are of type int)

tree.Put(1, "x") // 1->x
tree.Put(2, "b") // 1->x, 2->b (in order)
tree.Put(1, "a") // 1->a, 2->b (in order, replacement)
```